src/PaginatorFactory.php

```diff
@@ -37,6 +39,43 @@ public function withAdapter(AdapterInterface $adapter): Paginator
         );
     }
 
+    /**
+     * Generate a paginator with the given adapter wrapped in a caching adapter.
+     *
+     * You must configure the default cache or an exception will be thrown. It is also advisable to configure a default
+     * TTL, as it's unlikely that you'll want paginated items cached forever.
+     *
+     * The cache key prefix should be unique to the data set you are paginating. Cache key collisions are not considered
+     * so you are responsible for providing something that will prevent the cache from returning incorrect data across
+     * multiple paginators.
+     *
+     * @template TKey of array-key
+     * @template TValue
+     * @param AdapterInterface<TKey, TValue> $adapter
+     * @param non-empty-string $cacheKeyPrefix
+     * @return Paginator<TKey, TValue>
+     * @throws RuntimeException If a default cache has not been configured.
+     */
+    public function withCachingAdapter(AdapterInterface $adapter, string $cacheKeyPrefix): Paginator
```

Comment:
Expanding the factory is a simple and good idea, but I had been thinking about a delegator. That would allow to keep the fact that there is caching out of everything else.

Reply:
With some additional config we could delegate by default too

Reply:
Having thought about this some more @froschdesign, A delegator isn't going to work because the user _must_ provide a unique cache-key prefix for each different result set.

I don't think this is a big deal, because opting-in to caching where it makes sense in your application is better than enabling cache for _all_ pagers. A lot of the time caching is not necessary.